### ISSUE TITLE
[question] DDPG and VecNormalize wrapper

### ISSUE THREAD

**Issue by kosmylo** [hill-a/stable-baselines#902]
I have noticed that in DDPG we can have as parameters **normalize_returns** and **normalize_observations**. However, I normalize the observations and returns through **VecNormalize**.

```python
import os
import read_params
import matplotlib.pyplot as plt
import tensorflow as tf
import numpy as np

from environment import ChargingStation

from stable_baselines.ddpg.policies import MlpPolicy
from stable_baselines.common.vec_env import DummyVecEnv, VecNormalize
from stable_baselines.common.noise import NormalActionNoise, OrnsteinUhlenbeckActionNoise, AdaptiveParamNoiseSpec
from stable_baselines import DDPG
from stable_baselines.bench import Monitor
from stable_baselines import results_plotter

params, profiles = read_params.Charging_Station_Params()

# Create unique log dir
log_dir = "/tmp/ddpg/"
os.makedirs(log_dir, exist_ok = True)

env = ChargingStation()
env = Monitor(env, log_dir, allow_early_resets = True)
env = DummyVecEnv([lambda: env])

# Automatically normalize the input features and rewards and stack the previous observations
env = VecNormalize(env, norm_obs = True, norm_reward = True, clip_obs = 10.)

# the noise objects for DDPG
n_actions = env.action_space.shape[-1]
param_noise = AdaptiveParamNoiseSpec(initial_stddev=0.1, desired_action_stddev=0.1)
action_noise = None

# Custom MLP policy 
policy_kwargs = dict(act_fun = tf.nn.tanh, layers = [128, 128, 128])
buffer_size = 100000
gamma = 0.99

model = DDPG(MlpPolicy, env, gamma = gamma, policy_kwargs = policy_kwargs, buffer_size = buffer_size, verbose = 1, param_noise = param_noise, action_noise = action_noise)

model.learn(total_timesteps = params.time_steps)

# Don't forget to save the VecNormalize statistics when saving the agent
model.save(log_dir + "ddpg_ev_charging")
env.save(os.path.join(log_dir, "vec_normalize.pkl"))

# Plot learning curve
results_plotter.plot_results([log_dir], params.time_steps, results_plotter.X_TIMESTEPS, "DDPG ChargingStation")
plt.show()

```
Is there any advantage or disadvantage of using VecNormalize instead of the DDPG parameters?

**Comment by araffin** [comment]
Hello,
this might be a duplicate of https://github.com/hill-a/stable-baselines/issues/736

What is your question?

**Comment by kosmylo** [comment]
Thanks for the reply. It is still not clear to me if I have to set these parameters as True, even if I use VecNormalize or it is not necessary.

**Comment by araffin** [comment]
>It is still not clear to me if I have to set these parameters as True, even if I use VecNormalize or it is not necessary.

The two methods are doing the same thing, so if you activate one normalization, you should not use the other one.
They both exist for legacy reason... That's why we have only one way of normalizing features/returns in SB3.
